docs/docs/lists-conditional-rendering.md

```diff
@@ -0,0 +1,264 @@
+---
+id: forms
+title: Lists & Conditional Rendering
+permalink: docs/lists-conditional-rendering.html
+prev: 
+next: 
+---
+
+## Lists
+
+First, let's talk about transforming lists in Javascript.
+
+Given the code below, we use the `map()` function to take an array of `numbers`, double their values, and then output them to a new array `doubled`.
+
+```javascript
+const numbers = [1, 2, 3, 4, 5];
+numbers.map((number) => number * 2); // outputs: [2, 4, 6, 8 10]
+```
+
+In React, transforming arrays into lists of elements is nearly identical.
+
+### Basic List
+
+A component must return only a single item. If you are returning multiple elements, wrap them in `<div></div>`.
+
+Here we take our list of `numbers` and generate a collection of `<li>` elements. We are using the plain Javascript `map()` function. Learn more about it <a target="_blank" href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map">here</a>.
+
+```javascript
+class Numbers extends React.Component {
+  render() {
+    const numbers = [1, 2, 3, 4, 5];
+    return (
+      <div>
+        <ul>
+          {numbers.map((number) => <li>{number}</li>)}
+        </ul>
+      </div>
+    );
+  }
+}
+```
+
+<a target="_blank" href="https://codepen.io/ericnakagawa/pen/wzxEmv/?editors=0011">Try it out on Codepen.</a>
+
+If you run this code, you'll be given a warning that you should provide a key for list items. Keys are important and you should include them when creating lists of elements.
+
+Let's refactor this code to replace our `<li>` element with a [functional component]() and provide it a `key` and fix the missing key issue. 
+
+```javascript
+function Number(props) {
+  return <li>{props.value}</li>  
+}
+function Numbers(props) {
+  const numbers = [1, 2, 3, 4, 5];
+  return (
+    <div>
+      <ul>
+        {numbers.map((number, index) => <Number key={index} value={number} />)}
+      </ul>
+    </div>
+  );
+}
+```
+
+<a target="_blank" href="https://codepen.io/ericnakagawa/pen/gwjdzN?editors=0011">Try it out on Codepen.</a>
+
+
+### Keys
+
+In the previous example we provide the index of an item in our array as our `key`. It is important to always provide keys to your lists of elements.
+
+* For lists that are unlikely or won't change, using an item index is fine.
+* For lists that may change, it is important to use a unique key. A string representing identity of the data item can also work.
+
+* Keys should be unique amongst siblings. This means that the keys used with groups of related elements in a list should be unique .
+* You will be warned when you don't include keys.
+
+#### Performance
+
+In React, the absence of keys on an element can impact your application performance.
+
+When a `key` is provided, React chooses the most performant method for updating the rendered objects.
+
+#### Example 1
+
+Given the HTML below:
```

Comment:
Can we not use HTML in this document? It is irrelevant to React users and a bit confusing.